parcels/particleset.py

```diff
@@ -651,7 +653,7 @@ def execute(self, pyfunc=AdvectionRK4, endtime=None, runtime=None, dt=1.,
             if self.ptype.uses_jit:
                 self.kernel.remove_lib()
                 cppargs = ['-DDOUBLE_COORD_VARIABLES'] if self.lonlatdepth_dtype == np.float64 else None
-                self.kernel.compile(compiler=GNUCompiler(cppargs=cppargs))
+                self.kernel.compile(compiler=GNUCompiler(cppargs=cppargs, incdirs=[path.join(get_package_dir(), 'include'), "."]))
```

Comment:
It indeed requires the person writing a customized kernel interface to know what the arguments are. Given that this is an extremely rare case, as re-writing the Kernel class or subclassing it is not an everyday-casual thing, I think it's save to require it.

It is now that way because the `Compiler`-class interface can now also be used outside the code generator to JIT-compile C-code files and interlink them with the in-time code generation parts. Now, if one has pre-cooked C-source and header files and just wants to get them into `ctypes`, one doesn't require including the tools or code generator. Also, these codes are used in classes outside the kernel - such as Particles or Nodes. So, if we want to avoid compile ambiguity, we cannot compile every C-source with the parcels-include directory that the Kernel relies upon. And thus, every user of the compiler needs to explicitly state its `include` and `src` paths.

So, in short: in order to avoid strict dependencies allow allow for JIT-compiled support classes from other classes and packages, this move was inevitable.

Does it create a problem for your group ?

Reply:
Yeah, this is perfectly reasonable! I'm the one calling the kernel compilation directly in our case, so I can adjust for the new interface no problem.